internal/impl/postgresql/pglogicalstream/heartbeat.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Licensed as a Redpanda Enterprise file under the Redpanda Community
+// License (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+// https://github.com/redpanda-data/connect/v4/blob/main/licenses/rcl.md
+
+package pglogicalstream
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"github.com/redpanda-data/benthos/v4/public/service"
+
+	"github.com/redpanda-data/connect/v4/internal/asyncroutine"
+)
+
+type heartbeat struct {
+	db            *sql.DB
+	task          *asyncroutine.Periodic
+	logger        *service.Logger
+	prefix, value string
+}
+
+func newHeartbeat(dsn string, interval time.Duration, logger *service.Logger, prefix, value string) (*heartbeat, error) {
+	dbConn, err := openPgConnectionFromConfig(dsn)
+	if err != nil {
+		return nil, err
+	}
+	h := &heartbeat{dbConn, nil, logger, prefix, value}
```

Comment:
I personally prefer decoupling the composite literal from the struct definition by explicitly specifying field names but since this is a private type and very close to the instantiation, it doesn't seem as important, unless you all care about consistent style. 

https://google.github.io/styleguide/go/decisions.html#literal-formatting

Reply:
Yes agreed. I was just being lazy and going quickly, will cleanup. Thanks for the review!